shared/utility/strutil/strutil.hpp

```diff
@@ -29,97 +29,307 @@
 #define UTILITY_STRUTIL_HPP
 
 #include <algorithm>
+#include <functional>
+#include <iterator>
+#include <sstream>
 #include <string>
-
+#include <vector>
 
 /**
  * @author Trent Houliston
- * @author Alex Biddulph: added vector version of endsWith ans startsWith 01-06-2021
- * @author Alex Biddulph: added trimLeft, trimRight, trim, and removeAll functions 10-12-2012
+ * @author Alex Biddulph: added vector version of ends_with ans startsWith 01-06-2021
+ * @author Alex Biddulph: added trim_left, trim_right, trim, and remove_all functions 10-12-2012
  * @author Monica Olejniczak added split functions
  * @author Brendan Annable
  */
 namespace utility::strutil {
 
+    /**
+     * @brief Checks if a string ends with a specific substring.
+     *
+     * This function compares the given string `str` with the specified `ending` substring
+     * and determines whether `str` ends with `ending`. The comparison is case-sensitive.
+     *
+     * @param str The string to check for the ending substring.
+     * @param ending The substring to search for at the end of the string.
+     * @return `true` if `str` ends with `ending`, `false` otherwise.
+     */
     // http://stackoverflow.com/a/874160/1387006
```

Comment:
```suggestion
    /**
     * @brief Checks if a string ends with a specific substring.
     *
     * This function compares the given string `str` with the specified `ending` substring
     * and determines whether `str` ends with `ending`. The comparison is case-sensitive.
     * http://stackoverflow.com/a/874160/1387006
     *
     * @param str The string to check for the ending substring.
     * @param ending The substring to search for at the end of the string.
     * @return `true` if `str` ends with `ending`, `false` otherwise.
     */
```